python/cudf/cudf/tests/test_csv.py

```diff
@@ -1334,6 +1334,36 @@ def test_csv_reader_hexadecimals(pdf_dtype, gdf_dtype):
         assert_eq(pdf, gdf)
 
 
+@pytest.mark.parametrize(
+    "np_dtype, gdf_dtype",
+    [("int_", "hex"), ("int32", "hex32"), ("int64", "hex64")],
+)
+def test_csv_reader_hexadecimal_overflow(np_dtype, gdf_dtype):
+    # This tests values which cause an overflow warning that will become an
+    # error in pandas. NumPy wraps the overflow silently up to the bounds of a
+    # signed int64.
```

Comment:
is it always 64?

Reply:
Yup. By default, numpy treats types larger than int64/uint64 as `object` as the dtype and uses a Python `int` to back it. There are larger types, like `np.int128`, but they're not used by default. The wider the type, the less-wide the support, I guess.

Reply:
Correction: There are 128-bit float types (depending on the platform/build of NumPy) like `np.longdouble` but 128-bit integers are not in the NumPy API. However, 128-bit integers seem to be used internally and references can be found in a few places in the source:
- https://github.com/numpy/numpy/issues/9992
- https://github.com/numpy/numpy/search?q=int128&type=code